src/Events/EventDispatchableTrait.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace League\OAuth2\Server\Events;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+
+trait EventDispatchableTrait
+{
+    /** @var EventDispatcherInterface|null */
+    protected $eventDispatcher;
+
+    public function useEventDispatcher(?EventDispatcherInterface $eventDispatcher): void
+    {
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function dispatchEvent(object $event): object
```

Comment:
```suggestion
    protected function dispatchEvent(object $event): object
```

This method is only called as `$this->dispatchEvent()`, so there is no need for this method to be public.

Reply:
I guess I need to make `dispatchEvent` method private and remove from interface.